### DISCUSSION TITLE
Public npm package 401 error, why?

### DISCUSSION THREAD

**Discussion by punkbit** [community/community#23386] [npm]
<p>Hi,</p>
<p>Made a repository and related npm package public. On npm install there’s a 401 error.</p>
<pre><code class="lang-auto">Unable to authenticate, need: Basic realm="GitHub Package Registry"
</code></pre>
<p>The npmrc has the organisation.</p>
<p>Why is this happening if the repository and npm package is public?</p>

**Comment by punkbit** [comment, marked as answer]
<p>Github registry requires the user to be authenticated to Github using a private token via npm login with <code>repo</code> and <code>read:package</code> scopes, even for installing public packages.</p>

**Reply by CEbbinghaus** [reply]
Because that makes so much sense. How do they expect people to use github registry if it requires a github account just to pull a public package. I have very little understanding for the choice. I guess its created first and foremost aimed at private repositories / packages but forcing everyone that wants to use a public github registry package to also have a github account and set up a access token is unsustainable

**Reply by WozniakMac** [reply]
It's free so they want to know who is using it just in case if anyone will overuse it

**Reply by Haringat** [reply]
@WozniakMac Pretty much every other service knows how to create a rate limiting. I doubt that github would be incapable of doing so.

**Reply by freisenhauer** [reply]
Okay, thanks for this clear answer. This makes it absolutely unusable for us.
